EmbeddedEndpointSetter test failing

> @mcastorina This test is failing; not sure why.
>
> Edit: it looks like #3308 changed the logic in a way that subtly broke the test.
>
> ```
> --- FAIL: TestEmbeddedEndpointSetter (0.00s)
>     endpoint_customizer_test.go:12: 
>                 Error Trace:    /tmp/trufflehog/pkg/detectors/endpoint_customizer_test.go:12
>                 Error:          Not equal: 
>                                 expected: []string{"baz"}
>                                 actual  : []string(nil)
>                                 
>                                 Diff:
>                                 --- Expected
>                                 +++ Actual
>                                 @@ -1,4 +1,2 @@
>                                 -([]string) (len=1) {
>                                 - (string) (len=3) "baz"
>                                 -}
>                                 +([]string) <nil>
>                                  
>                 Test:           TestEmbeddedEndpointSetter
> FAIL
> ```
> https://github.com/trufflesecurity/trufflehog/blob/bf6f3136971c9d21c6ef5a6d9bd1d11520102c30/pkg/detectors/endpoint_customizer_test.go#L12

_Originally posted by @rgmz in https://github.com/trufflesecurity/trufflehog/issues/3773#issuecomment-2541588054_


----

Unclear whether the feature is broken or the test is broken.

--- a/pkg/detectors/endpoint_customizer_test.go
+++ b/pkg/detectors/endpoint_customizer_test.go
@@ -9,8 +9,19 @@ import (
 func TestEmbeddedEndpointSetter(t *testing.T) {
 	type Scanner struct{ EndpointSetter }
 	var s Scanner
+	// use foundendpoint to set the "baz" in endpoints
+	s.useFoundEndpoints = true
 	assert.Equal(t, []string{"baz"}, s.Endpoints("baz"))
+	// setting "foo" and "bar" as configured endpoint
 	assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
+	// return error as no endpoints are passed
 	assert.Error(t, s.SetConfiguredEndpoints())
-	assert.Equal(t, []string{"foo", "bar"}, s.Endpoints("baz"))
+	// as use foundendpoint is true, "baz" will be added in the endpoints list
+	assert.Equal(t, []string{"foo", "bar", "baz"}, s.Endpoints("baz"))
+	// setting cloudendpoints
+	s.useCloudEndpoint = true
+	// cloudendpoint must be set in order to be added in the list along with useCloudEndpoint as true
+	s.cloudEndpoint = "test"
+	// reason for "foo" and "bar" here is that they are already configured
+	assert.Equal(t, []string{"foo", "bar", "test"}, s.Endpoints())
 }

--- a/pkg/detectors/endpoint_customizer_test.go
+++ b/pkg/detectors/endpoint_customizer_test.go
@@ -8,29 +8,53 @@ import (
 
 func TestEmbeddedEndpointSetter(t *testing.T) {
 	type Scanner struct{ EndpointSetter }
+
 	var s Scanner
-	// set useFoundEndpoints to true to add the "baz" in endpoints
-	s.useFoundEndpoints = true
-	assert.Equal(t, []string{"baz"}, s.Endpoints("baz"))
-	// setting "foo" and "bar" as configured endpoint
-	assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
-	// return error as no endpoints are passed
-	assert.Error(t, s.SetConfiguredEndpoints())
-	// as useFoundEndpoints is true, "baz" will be added in the endpoints list
-	assert.Equal(t, []string{"foo", "bar", "baz"}, s.Endpoints("baz"))
-	// setting cloudEndpoint along with setting useCloudEndpoint as true
-	s.useCloudEndpoint = true
-	// cloudEndpoint must be set in order to be added in the list along with useCloudEndpoint as true
-	s.cloudEndpoint = "test"
-	// reason for "foo" and "bar" here is that they are already configured
-	assert.Equal(t, []string{"foo", "bar", "test"}, s.Endpoints())
-	// set useFoundEndpoints and useCloudEndpoint to false
-	s.useFoundEndpoints = false
-	s.useCloudEndpoint = false
-	// as both useFoundEndpoints and useCloudEndpoint are set to false, passing any endpoint to Endpoints() will not be added
-	assert.Equal(t, []string{"foo", "bar"}, s.Endpoints("test"))
-	// set any new endpoint as cloudEndpoint
-	s.cloudEndpoint = "new"
-	// this time it will not be added in list because useCloudEndpoint is set to false
-	assert.Equal(t, []string{"foo", "bar"}, s.Endpoints())
+
+	t.Run("useFoundEndpoints is true", func(t *testing.T) {
+		s.useFoundEndpoints = true
+
+		// "baz" is passed to Endpoints, should appear in the result
+		assert.Equal(t, []string{"baz"}, s.Endpoints("baz"))
+	})
+
+	t.Run("setting configured endpoints", func(t *testing.T) {
+		// Setting "foo" and "bar"
+		assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
+
+		// Returning error because no endpoints are passed
+		assert.Error(t, s.SetConfiguredEndpoints())
+	})
+
+	// "foo" and "bar" are added as configured endpoint
+
+	t.Run("useFoundEndpoints adds new endpoints", func(t *testing.T) {
+		// "baz" is added because useFoundEndpoints is true
+		assert.Equal(t, []string{"foo", "bar", "baz"}, s.Endpoints("baz"))
+	})
+
+	t.Run("useCloudEndpoint is true", func(t *testing.T) {
+		s.useCloudEndpoint = true
+		s.cloudEndpoint = "test"
+
+		// "test" is added because useCloudEndpoint is true and cloudEndpoint is set
+		assert.Equal(t, []string{"foo", "bar", "test"}, s.Endpoints())
+	})
+
+	t.Run("disable both foundEndpoints and cloudEndpoint", func(t *testing.T) {
+		// now disable both useFoundEndpoints and useCloudEndpoint
+		s.useFoundEndpoints = false
+		s.useCloudEndpoint = false
+
+		// "test" won't be added
+		assert.Equal(t, []string{"foo", "bar"}, s.Endpoints("test"))
+	})
+
+	t.Run("cloudEndpoint not added when useCloudEndpoint is false", func(t *testing.T) {
+		s.cloudEndpoint = "new"
+
+		// "new" is not added because useCloudEndpoint is false
+		assert.Equal(t, []string{"foo", "bar"}, s.Endpoints())
+	})
+
 }

--- a/pkg/detectors/endpoint_customizer_test.go
+++ b/pkg/detectors/endpoint_customizer_test.go
@@ -9,19 +9,28 @@ import (
 func TestEmbeddedEndpointSetter(t *testing.T) {
 	type Scanner struct{ EndpointSetter }
 	var s Scanner
-	// use foundendpoint to set the "baz" in endpoints
+	// set useFoundEndpoints to set the "baz" in endpoints
 	s.useFoundEndpoints = true
 	assert.Equal(t, []string{"baz"}, s.Endpoints("baz"))
 	// setting "foo" and "bar" as configured endpoint
 	assert.NoError(t, s.SetConfiguredEndpoints("foo", "bar"))
 	// return error as no endpoints are passed
 	assert.Error(t, s.SetConfiguredEndpoints())
-	// as use foundendpoint is true, "baz" will be added in the endpoints list
+	// as useFoundEndpoints is true, "baz" will be added in the endpoints list
 	assert.Equal(t, []string{"foo", "bar", "baz"}, s.Endpoints("baz"))
-	// setting cloudendpoints
+	// setting cloudEndpoint along with setting useCloudEndpoint as true
 	s.useCloudEndpoint = true
-	// cloudendpoint must be set in order to be added in the list along with useCloudEndpoint as true
+	// cloudEndpoint must be set in order to be added in the list along with useCloudEndpoint as true
 	s.cloudEndpoint = "test"
 	// reason for "foo" and "bar" here is that they are already configured
 	assert.Equal(t, []string{"foo", "bar", "test"}, s.Endpoints())
+	// set useFoundEndpoints and useCloudEndpoint to false
+	s.useFoundEndpoints = false
+	s.useCloudEndpoint = false
+	// as both useFoundEndpoints and useCloudEndpoint are set to false, passing any endpoint to Endpoints() will not be added
+	assert.Equal(t, []string{"foo", "bar"}, s.Endpoints("test"))
+	// set any new endpoint as cloudEndpoint
+	s.cloudEndpoint = "new"
+	// this time it will not be added in list because useCloudEndpoint is set to false
+	assert.Equal(t, []string{"foo", "bar"}, s.Endpoints())
 }

--- a/pkg/detectors/endpoint_customizer_test.go
+++ b/pkg/detectors/endpoint_customizer_test.go
@@ -9,7 +9,7 @@ import (
 func TestEmbeddedEndpointSetter(t *testing.T) {
 	type Scanner struct{ EndpointSetter }
 	var s Scanner
-	// set useFoundEndpoints to set the "baz" in endpoints
+	// set useFoundEndpoints to true to add the "baz" in endpoints
 	s.useFoundEndpoints = true
 	assert.Equal(t, []string{"baz"}, s.Endpoints("baz"))
 	// setting "foo" and "bar" as configured endpoint

